src/Microsoft.OData.Client/ALinq/AddNewEndingTokenVisitor.cs

```diff
@@ -47,7 +47,7 @@ public void Visit(NonSystemToken tokenIn)
             {
                 if (newTokenToAdd != null)
                 {
-                    tokenIn.SetNextToken(newTokenToAdd);

```

Comment:
Why this change? Are you planning to remove the internal SetNextToken in favor of simply setting the public property?